hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -339,6 +339,7 @@ protected void processResult(CommitKeyRequest commitKeyRequest,
       // versioning of keys. So, this can be revisited later.
       if (omKeyInfo.getKeyLocationVersions().size() == 1) {
         omMetrics.incNumKeys();
+        omMetrics.incDataCommittedBytes(omKeyInfo.getDataSize());
```

Comment:
I think this will not survive restarts! @avijayanhwx?
Possibly might need to read it back from store if this is meant to be cumulative value.

Reply:
And also when key is overwrite, do we want to consider that also here.

Reply:
For numKeys we do best effort of writing a file in shutdown hook. (If unclean shutdown, numbers will not be accurate)
``    shutdownHook = () -> {
      saveOmMetrics();
    };
``

Reply:
Maybe a separate metric for tracking overwrites? Although overwriting use case for this particular metric should probably still add to the cumulative value. Tracking effective committed data accounting for overwrites cannot be done by metrics system I think, maybe some like Recon can calculate this after the fact.

Reply:
Like @bharatviswa504 said, OM has a best effort "save metrics" logic to allow it to continue counters from last stop. But, that does not save us from ungraceful shutdowns. 

If our goal is to understand & analyze rate of growth of data in the cluster, I believe we don't need to worry about overwrites (since we can treat as overwrites also as "new load" on the cluster). If we want to track the actual data that is currently being managed by Ozone, then I would suggest using Recon. This is already being done for FSO data in HDDS-5305, and can be queried at volume/bucket/dir level.

Reply:
If the goal is just to track as said above, I am fine to not complicate things with key override.

Reply:
This line should be outside the if block, as we want to increment for any scenario new/override key.

Reply:
Sorry that I did not pay attention to this earlier. The goal here is to get a trend of data written into the cluster.

Reply:
Found that this will be an important config in debugging issues in the cluster.

Reply:
Addressed @bharatviswa504 comment